test_runner/src/main/kotlin/ftl/args/AndroidArgs.kt

```diff
@@ -91,24 +91,27 @@ AndroidArgs
     }
 }
 
+// Changes these based on type
 val AndroidArgs.isDontAutograntPermissions
     get() = !(grantPermissions.isNotNull() && (grantPermissions.equals("all")))
 
 val AndroidArgs.isInstrumentationTest
-    get() = appApk.isNotNull() &&
-            testApk.isNotNull() ||
-            additionalAppTestApks.isNotEmpty() &&
-            (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() })
+    get() = if (type != null) (type == Type.INSTRUMENTATION) else
+        (appApk.isNotNull() && testApk.isNotNull() || additionalAppTestApks.isNotEmpty() &&
+            (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() }))
```

Comment:
IMHO  with this formatting this is not so readable I would use something like
```kotlin
get() = if (type != null) (type == Type.INSTRUMENTATION) else validateApks()

fun AndroidArgs.validateApks() = (appApk.isNotNull() && testApk.isNotNull() || 
    additionalAppTestApks.isNotEmpty() &&
    (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() }))
```

Reply:
👍